include/eosio/opaque.hpp

```diff
@@ -86,9 +88,13 @@ class opaque<std::vector<T>> : public opaque_base<std::vector<T>> {
  public:
    using opaque_base<std::vector<T>>::opaque_base;
 
-   size_t unpack_size() {
-      uint32_t num;
-      varuint32_from_bin(num, this->bin);
+   /** Determine the size of the vector to be unpacked.
+
+    @pre this->empty() should be false.
+   */
+   uint64_t unpack_size() {
```

Comment:
Could you also add the precondition doc for opaque_base<T>::unpack()?

Reply:
Done